service/sharddistributor/leader/process/processor.go

```diff
@@ -390,9 +385,14 @@ func (p *namespaceProcessor) rebalanceShardsImpl(ctx context.Context, metricsLoo
 		return nil
 	}
 
-	p.addAssignmentsToNamespaceState(namespaceState, currentAssignments)
-	p.logger.Info("Applying new shard distribution.")
+	newState := p.getNewAssignmentsState(namespaceState, currentAssignments)
+	if p.dynamicCfg.GetMigrationMode(p.namespaceCfg.Name) != types.MigrationModeONBOARDED {
+		p.logger.Info("Running rebalancing in shadow mode", tag.Dynamic("old_assignments", namespaceState.ShardAssignments), tag.Dynamic("new_assignments", newState))
+		return nil
```

Comment:
I think we should move `metricsLoopScope.UpdateGauge(metrics.ShardDistributorActiveShards, float64(totalActiveShards))` from below to somewhere upper - to not miss this metric in case of shadow mode.